cmd/fscrypt/keys.go

```diff
@@ -179,6 +181,19 @@ func makeKeyFunc(supportRetry, shouldConfirm bool, prefix string) actions.KeyFun
 			if prefix != "" {
 				return nil, ErrNotPassphrase
 			}
+			if keyFileFlag.Value == "" { // Reading from STDIN
+				in := bufio.NewReader(os.Stdin)
+				buf := make([]byte, 0, 32)
+				for i := 1; i <= 32; i++ {
+					c, err := in.ReadByte()
+					buf = append(buf, c)
+					if err == io.EOF {
+						break
+					}
+				}
+				key := bytes.NewReader(buf)
+				return crypto.NewFixedLengthKeyFromReader(key, metadata.InternalKeyLen)
```

Comment:
Why not just:

    return crypto.NewFixedLengthKeyFromReader(bufio.NewReader(os.Stdin), metadata.InternalKeyLen)